src/sfdt/sfdtHelpers/listHelpers.ts

```diff
@@ -0,0 +1,227 @@
+ * 3. if there is styleName, check in the styles array in sfdt; iterate through the basedOn until you get stopping format
+ * getCompleteParaFormat iterates through styles to return the current updated paragraphFormat which has the required listFormat
+ * @param paraBlock paragraph block
+ * @param stylesArray sfdt styles array
+ */
+export const getListFormatOfParaFormat = (paraBlock: BlockType, stylesArray: StyleType[]) => {
+	const paraFormat = get(paraBlock, 'paragraphFormat');
+	// Check for listFormat. If default, return it
+	const listFormat = get(paraFormat, 'listFormat');
+	// if listFormat.listId = -1: the data is in the block, but is not a running list; so no need to update with the paragraphFormat styles
+	// check https://github.com/syncfusion/ej2-javascript-ui-controls/blob/d19a2fcd57969a9dfb7986c02806984b873d8ff4/controls/documenteditor/src/document-editor/implementation/format/list-format.ts#L79
+
+	if (listFormat && listFormat.listId < 0) {
+		return listFormat;
+	}
+	// if not invalid list, then get the complete paraformat
+	const updatedParaFormat = getCompleteParaFormat(paraBlock, stylesArray);
+	// Compare with defaultListFormat
+	return Object.assign({}, defaultListFormat, get(updatedParaFormat, 'listFormat'));
+};
+
+/**
+ * Get updated listFormat number in respect to sfdt updates
+ * This class just performs the calculation of listLevel from the sfdt and parse the list number text for that. For parsing listLevels as well: https://github.com/syncfusion/ej2-javascript-ui-controls/blob/b2424abb138987c42f77b67a02bd933aaccb9629/controls/documenteditor/src/document-editor/implementation/viewer/layout.ts#L1947
+ * @param sfdt whole sfdt data
+ */
+export function ParseListLevel(sfdt: SFDTType) {
+	this.sfdt = sfdt;
+	this.list = {};
+	this.listLevelNumber = 0;
+	/**
+	 * renderedLists stores the processed abstractList object (by abstractListId) with its count
+	 * Since the keys are object and we don't need to iterate them (just check if the current abstractListId from the listId is processed or not), use WeakMap instead of object/Map
+	 * This map is used to count the number of times the abstractList is processed, to get the list number with the levels
+	 */
+	this.renderedLists = new WeakMap();
+	// clear the renderedLists stored
+	this.clearLists = function() {
+		this.renderedLists = new WeakMap();
+	};
+
+	this.setList = function(list: ListType, listLevelNumber: number): void {
+		this.list = list;
+		this.listLevelNumber = listLevelNumber;
+	};
+	/**
+	 * Get list by listId (from lists array in sfdt)
+	 * @param listId listFormat listId from lists array
+	 */
+	this.getListById = function(listId: number): ListType {
+		const lists = get(this.sfdt, 'lists');
+		if (isNullOrUndefined(lists)) {
+			return undefined;
+		}
+
+		for (const element of lists) {
+			if (!isNullOrUndefined(element) && get(element, 'listId') === listId) {
+				return element;
+			}
+		}
+		return undefined;
+	};
```

Comment:
The signature doesnt match the return type. The annotation says `getListById` returns `ListType` but the code shows it returns either `ListType` or `undefined`

Reply:
Same for many methods so add proper type signature to them.